src/net/sourceforge/kolmafia/objectpool/Concoction.java

```diff
@@ -182,6 +182,49 @@ public Concoction(final String name, final String property) {
     this.setEffectName();
   }
 
+  // Copy constructor primarily for testing.  Shallow copy.
+  public Concoction(Concoction source) {
```

Comment:
Is this code you were writing to test if cloning breaks it? Did you mean to push it?

Reply:
WIP.  Give me 15 minutes.

Reply:
OK.  I implemented a copy constructor because that seemed more robust than making Concoction cloneable.  I wrote a new test comparing the clone and original.  compareTo failed when comparing.  They should be equal and were not.  I then put my previous fix in (but left yours there) uncommented the failing tests, saw they passed, delinted the tests and confirmed checkconcotion works.

Reply:
@jaadams5 I've again removed the line below - I told you it doesn't make sense! I've replaced the == with a .equals() as I suggested many hours ago and this contrived test now passes. Once you've seen that it does, I'd like to remove this function and the associated test as they're not meaningful

Reply:
Also please don't add commits to a branch someone else is working on - you can just make a branch from this branch and add them there.

Reply:
First an apology for committing.  My model of how to collaborate on GitHub continues to evolve.

As for the rest, you are right and I figured it out.  First I had missed the fact that Concoctions overrides equals and so the comparison was looking at name and an id and not just asking "are they the same object?".  Second I did have a failing test with only your change.  It was contrived under an incorrect assumption about equals but it failed.  When I tried to reproduce it after your correct assertion that I was missing something, I failed to do so.  That was operator error, presumably due to multi-tasking :-)  In any event the next step is to back out what I did, confirm checkconcoctions and tests work and approve it.